From 82d4f09a97a7410556f9b0e3a1012a7dcb974f30 Mon Sep 17 00:00:00 2001
From: juntingwang <Junting.Wang@windriver.com>
Date: Mon, 8 Jul 2013 17:17:13 +0800
Subject: HW keyboard is not functional in Pin view

Pin view for screen lock can accept key events, the issue is focus is
not on the Pin view.
Modify pinentry to a EditText, so that it can always try to get focus.
And let security view request focus when resumed.

Due to the current lack of interest in laptop-type devices
on AOSP, this patch is not recommended for upstreaming.

Category: Device Enablement
Domain: IO
Upstream-Candidate: no

Issue: AXIA-2836
Change-Id: I94bc363542485bce47274f95442840e35404e20b
Signed-off-by: juntingwang <Junting.Wang@windriver.com>

Conflicts:

	core/res/res/layout/keyguard_pin_view.xml
	policy/src/com/android/internal/policy/impl/keyguard/KeyguardHostView.java
---
 packages/Keyguard/res/layout/keyguard_pin_view.xml |    2 +-
 .../src/com/android/keyguard/KeyguardHostView.java |    3 +--
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/packages/Keyguard/res/layout/keyguard_pin_view.xml b/packages/Keyguard/res/layout/keyguard_pin_view.xml
index 00c6a21..40136c0 100644
--- a/packages/Keyguard/res/layout/keyguard_pin_view.xml
+++ b/packages/Keyguard/res/layout/keyguard_pin_view.xml
@@ -47,7 +47,7 @@
           android:orientation="horizontal"
           android:layout_weight="1"
           >
-          <TextView android:id="@+id/pinEntry"
+          <EditText android:id="@+id/pinEntry"
                android:editable="true"
                android:layout_width="0dip"
                android:layout_height="match_parent"
diff --git a/packages/Keyguard/src/com/android/keyguard/KeyguardHostView.java b/packages/Keyguard/src/com/android/keyguard/KeyguardHostView.java
index 1bae9b8..8a9eca5 100644
--- a/packages/Keyguard/src/com/android/keyguard/KeyguardHostView.java
+++ b/packages/Keyguard/src/com/android/keyguard/KeyguardHostView.java
@@ -467,6 +467,7 @@ public class KeyguardHostView extends KeyguardViewBase {
 
     private void updateSecurityView(View view) {
         if (view instanceof KeyguardSecurityView) {
+            view.requestFocus();
             KeyguardSecurityView ksv = (KeyguardSecurityView) view;
             ksv.setKeyguardCallback(mCallback);
             ksv.setLockPatternUtils(mLockPatternUtils);
@@ -1041,8 +1042,6 @@ public class KeyguardHostView extends KeyguardViewBase {
         if (mViewStateManager != null) {
             mViewStateManager.showUsabilityHints();
         }
-
-        requestFocus();
     }
 
     @Override
-- 
1.7.4.1

